src/radar.cpp

```diff
@@ -777,3 +815,157 @@ void radarColour(UDWORD tileNumber, uint8_t r, uint8_t g, uint8_t b)
 	tileColours[tileNumber].byte.b = b;
 	tileColours[tileNumber].byte.a = 255;
 }
+
+
+// MARK: - RadarWidget
+
+std::shared_ptr<WIDGET> getRadarWidget()
+{
+	return pRadarWidget;
+}
+
+RadarWidget::RadarWidget()
+:  WIDGET()
+{ }
+
+#define WKEY_ORDER (getRightClickOrders()?WKEY_SECONDARY:WKEY_PRIMARY)
+#define WKEY_SELECT (getRightClickOrders()?WKEY_PRIMARY:WKEY_SECONDARY)
+#define MOUSE_SELECT (getRightClickOrders()?MOUSE_LMB:MOUSE_RMB)
+
+void RadarWidget::run(W_CONTEXT *)
+{
+	int PosX, PosY;
+	UDWORD	temp1, temp2;
+
+	if (m_hasHighlight)
+	{
+		// process drag events
+		if (mouseDrag(MOUSE_SELECT, &temp1, &temp2))
+		{
+			if (!getRotActive())
+			{
+				int x = mouseX();
+				int y = mouseY();
+				CalcRadarPosition(x, y, &PosX, &PosY);
+				setViewPos(PosX, PosY, true);
+				m_handlingDrag = true;
+				if (ctrlShiftDown())
+				{
+					playerPos.r.y = 0;
+				}
+			}
+		}
+		else
+		{
+			m_handlingDrag = false;
+		}
+	}
+	else
+	{
+		if (m_handlingDrag && !mouseDrag(MOUSE_SELECT, &temp1, &temp2))
+		{
+			m_handlingDrag = false;
+		}
+	}
+}
+
+void RadarWidget::display(int xOffset, int yOffset)
+{
+	if (!radarVisible()) { return; }
```

Comment:
```suggestion
	if (!radarVisible())
	{
		return;
	}
```
to make formatting consistent with our current guidelines.